internal/core/data/reading.go

```diff
@@ -72,10 +73,11 @@ func readingHandler(w http.ResponseWriter, r *http.Request) {
 				return
 			}
 
-			valid, err := isValidValueDescriptor(vd, reading)
-			if !valid {
-				http.Error(w, "Validation failed", http.StatusConflict)
-				LoggingClient.Error("Validation failed")
+			err = isValidValueDescriptor(vd, reading)
+			if err != nil {
+				err = errors.NewErrValueDescriptorInvalid(vd.Name, err)
```

Comment:
Maybe, instead of creating the ErrValueDescriptorInvalid here that isValidValueDescriptor return it directly?

Reply:
Done!